src/block.js

```diff
@@ -68,6 +84,20 @@ class Block {
       block.transactions.push(tx)
     }
 
+    // This Block contains a witness commit
+    if (block.hasWitnessCommit()) {
+      // The merkle root for the witness data is in an OP_RETURN output.
+      // There is no rule for the index of the output, so use filter to find it.
+      // The root is prepended with 0xaa21a9ed so check for 0x6a24aa21a9ed
+      // If multiple commits are found, the output with highest index is assumed.
+      let witnessCommits = block.transactions[0].outs
+        .filter(out => out.script.slice(0, 6).equals(Buffer.from('6a24aa21a9ed', 'hex')))
+        .map(out => out.script.slice(6, 38))
+
+      // Use the commit with the highest output (should only be one though)
+      block.witnessCommit = witnessCommits[witnessCommits.length - 1]
```

Comment:
I should probably check that witnessCommits.length > 0 here and throw an Error if it is.

Since having a witness for the first Transaction but not having the witness commit anywhere in the outputs would mean an invalid block.